### ISSUE TITLE
sing-box geosite export/lookup fail on 1.10.0-alpha.22/23

### ISSUE THREAD

**Issue by simplerick-simplefun** [SagerNet/sing-box#1961]
### Operating system

Linux

### System version

Ubuntu 22.04.4 LTS

### Installation type

Original sing-box Command Line

### Version

```shell
sing-box version 1.10.0-alpha.22

Environment: go1.22.5 linux/amd64
Tags: with_gvisor,with_quic,with_dhcp,with_wireguard,with_ech,with_utls,with_reality_server,with_acme,with_clash_api
Revision: 08afcdb2202afda9261f5aec1e712a1698976a32
CGO: disabled

sing-box version 1.10.0-alpha.22

Environment: go1.22.5 windows/amd64
Tags: with_gvisor,with_quic,with_dhcp,with_wireguard,with_ech,with_utls,with_reality_server,with_acme,with_clash_api
Revision: 08afcdb2202afda9261f5aec1e712a1698976a32
CGO: disabled
```

### Description

`sing-box geosite export <category>` fails on every "category" with `FATAL[0000] code <category> not exists!`

`sing-box geosite lookup <domain>` fails on every "domain" with `FATAL[0000] slice length: EOF`(Windows) `FATAL[0000] seek geosite.db: invalid argument`(Linux)

Ref: Same issue with https://github.com/SagerNet/sing-box/issues/1960

### Reproduction

`sing-box geosite export netflix`

`sing-box.exe geosite lookup 'baidu.com'`

### Logs

```shell
sing-box geosite export netflix
FATAL[0000] code netflix not exists!
```

Windows:
```
sing-box.exe geosite lookup "baidu.com"
FATAL[0000] slice length: EOF
```

Linux:
```
sing-box.exe geosite lookup 'baidu.com'
FATAL[0000] seek geosite.db: invalid argument
```

### Supporter

- [ ] I am a [sponsor](https://github.com/sponsors/nekohasekai/)

### Integrity requirements

- [X] I confirm that I have read the documentation, understand the meaning of all the configuration items I wrote, and did not pile up seemingly useful options or default values.
- [X] I confirm that I have provided the server and client configuration files and process that can be reproduced locally, instead of a complicated client configuration file that has been stripped of sensitive data.
- [X] I confirm that I have provided the simplest configuration that can be used to reproduce the error I reported, instead of depending on remote servers, TUN, graphical interface clients, or other closed-source software.
- [X] I confirm that I have provided the complete configuration files and logs, rather than just providing parts I think are useful out of confidence in my own intelligence.

**Comment by nekohasekai** [comment]
Should be fixed alpha 24.

**Comment by simplerick-simplefun** [comment]
fixed on 1.10.0-alpha.24
issue closed.
